extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/HttpUpgradeCheck.java

```diff
@@ -43,12 +43,22 @@ default boolean appliesTo(String endpointId) {
         return true;
     }
 
-    /**
-     * @param httpRequest {@link HttpServerRequest}; the HTTP 1.X request employing the 'Upgrade' header
-     * @param securityIdentity {@link SecurityIdentity}; the identity is null if the Quarkus Security extension is absent
-     * @param endpointId {@link WebSocket#endpointId()}
-     */
-    record HttpUpgradeContext(HttpServerRequest httpRequest, Uni<SecurityIdentity> securityIdentity, String endpointId) {
+    interface HttpUpgradeContext {
```

Comment:
Can you please clarify the idea behind making it an interface ?

Reply:
I need `RoutingContext` available because that is what eager security interceptors work with and users only want to use accessors `httpRequest()`, `securityIdentity()`, `endpoint()`, so users are not affected by this change.

Reply:
hmm, I could drop this change if it was important for you, it can be dropped once I manage to link endpoints based on path before authentication (even proactive authentication) happens. I don't think it is important, but I am open to revert this and reworked this PR if you see it as important. Thank you

Reply:
(that's PR after this one)

Reply:
sorry, one last comment :v: , I'd like to keep this change unless there is an issue, because it gives access to `RoutingContext`.

Reply:
Sure, I was just curious